src/verbs/queue_pair.rs

```diff
@@ -20,6 +19,62 @@ use super::{
     protection_domain::ProtectionDomain, AccessFlags,
 };
 
+#[cfg(feature = "debug")]
+use crate::verbs::address::Gid;
+
+#[derive(Debug, thiserror::Error)]
+#[non_exhaustive]
+pub enum ModifyQueuePairError {
+    #[error("modify queue pair failed")]
+    GenericError(#[from] io::Error),
+    #[cfg(feature = "debug")]
+    #[error("invalid transition from {cur_state:?} to {next_state:?}")]
+    InvalidTransition {
+        cur_state: QueuePairState,
+        next_state: QueuePairState,
+        source: io::Error,
+    },
+    #[cfg(feature = "debug")]
+    #[error("invalid transition from {cur_state:?} to {next_state:?}, possible invalid masks {invalid:?}, possible needed masks {needed:?}")]
+    InvalidAttributeMask {
+        cur_state: QueuePairState,
+        next_state: QueuePairState,
+        invalid: QueuePairAttributeMask,
+        needed: QueuePairAttributeMask,
+        source: io::Error,
+    },
+    #[cfg(feature = "debug")]
+    #[error("resolve route timed out, source gid index: {sgid_index}, destination gid: {gid}")]
+    ResolveRouteTimedout {
+        sgid_index: u8,
+        gid: Gid,
+        source: io::Error,
+    },
+    #[cfg(feature = "debug")]
+    #[error("network unreachable, source gid index: {sgid_index}, destination gid: {gid}")]
+    NetworkUnreachable {
+        sgid_index: u8,
+        gid: Gid,
+        source: io::Error,
+    },
+}
+
+#[derive(Debug, thiserror::Error)]
+#[non_exhaustive]
+pub enum PostSendError {
+    #[error("post send failed")]
+    GenericError(#[from] io::Error),
+    #[cfg(feature = "debug")]
+    #[error("invalid value provided in work request")]
+    InvalidWorkRequest(#[source] io::Error),
+    #[cfg(feature = "debug")]
+    #[error("invalid value provided in queue pair")]
+    InvalidValueInQueuePair(#[source] io::Error),
```

Comment:
`InvalidQueuePair` may be a better name for this error.

Reply:
Done